### PR TITLE
offchain - remove unused lp reader + minor fixes

### PR DESCRIPTION
**Pull request by dimkouv**
- delete log poller reader
- use `logpoller.Unconfirmed` instead of `logpoller.Confirmations(0)`
- move `ParseLogs` func

### CODE REVIEW

File: core/services/ocr2/plugins/ccip/internal/ccipdata/reader.go [hunk 1]
```diff
@@ -36,5 +38,28 @@ func LogsConfirmations(finalized bool) logpoller.Confirmations {
...
 	}
-	return logpoller.Confirmations(0)
+	return logpoller.Unconfirmed
+}
+
+func ParseLogs[T any](logs []logpoller.Log, lggr logger.Logger, parseFunc func(log types.Log) (*T, error)) ([]Event[T], error) {
+	reqs := make([]Event[T], 0, len(logs))
+	for _, log := range logs {
+		data, err := parseFunc(log.ToGethLog())
+		if err == nil {
```
Comment (jarnaud) [inline core/services/ocr2/plugins/ccip/internal/ccipdata/reader.go:48]: should we keep the errors to be displayed in the warning below?

**Review by jarnaud — APPROVED** [review]

File: core/services/ocr2/plugins/ccip/internal/ccipdata/reader.go (hunk 1 above)
Comment (dimkouv) [reply core/services/ocr2/plugins/ccip/internal/ccipdata/reader.go:48]: this looks weird indeed, i'll open a ticket to investigate if it's safe to return an error.

File: core/services/ocr2/plugins/ccip/internal/ccipdata/logpoller_test.go [hunk 2]
```diff
@@ -1,69 +0,0 @@
...
-	assert.True(t, offRamp == offRamp2)
-
-	priceReg, err := c.loadPriceRegistry(someAddr)
-	assert.NoError(t, err)
-	priceReg2, err := c.loadPriceRegistry(someAddr)
-	assert.NoError(t, err)
-	assert.True(t, priceReg == priceReg2)
-}
-
-func Test_parseLogs(t *testing.T) {
```
Comment (mateusz-sekara) [inline core/services/ocr2/plugins/ccip/internal/ccipdata/logpoller_test.go:42]: I think we should keep that test, no?

Comment (dimkouv) [reply core/services/ocr2/plugins/ccip/internal/ccipdata/logpoller_test.go:42]: nice catch

**Pull request merged**